packages/xod-client/src/editor/actions.js

```diff
@@ -197,6 +198,50 @@ export const setCurrentPatchOffset = newOffset => ({
   payload: newOffset,
 });
 
+const setPatchOffsetByEntitiesBoundingBox = getDesiredOffset => (dispatch, getState) => {
+  const state = getState();
+  const maybeCurrentPatchPath = Selectors.getCurrentPatchPath(state);
+  if (maybeCurrentPatchPath.isNothing) return;
+
+  const currentPatchPath = explodeMaybe(
+    'Imposible error: No currentPatchPath to autopan',
+    maybeCurrentPatchPath
+  );
+  const project = ProjectSelectors.getProject(state);
+  const currentPatch = XP.getPatchByPathUnsafe(currentPatchPath, project);
+
+  const entitiesBoundingBox = getBoundingBox(
+    currentPatch,
+    project,
+    {
+      nodes: XP.listNodes(currentPatch),
+      comments: XP.listComments(currentPatch),
+    }
+  );
+
+  // TODO: too dirty?
+  const patchBoundingRect = document.querySelector('.PatchWrapper-container').getBoundingClientRect();
```

Comment:
I think so. Imagine, how would you unit-test the action if you’d have to? There will be no DOM. Actions should not touch presentation.

A possible solution is to add viewport rect property to Redux state and oblige the component who owns `.PatchWrapper-container` to keep it up to date.